examples/Cassie/run_dircon_squatting.cc

```diff
@@ -0,0 +1,817 @@
+DEFINE_string(data_directory, "../dairlib_data/cassie_trajopt_data/",
+              "directory to save/read data");
+DEFINE_int32(max_iter, 100000, "Iteration limit");
+DEFINE_double(duration, 0.4, "Duration of the single support phase (s)");
+DEFINE_double(tol, 1e-4,
+              "Tolerance for constraint violation and dual gap");
+
+namespace dairlib {
+
+/// Trajectory optimization of fixed-spring cassie squating
+/// With the default initial guess, the solving time is about 100 seconds.
+
+// Constraint to fix the position of a point on a body (for initial guess)
+class BodyPonitPositionConstraint : public DirconAbstractConstraint<double> {
+ public:
+  BodyPonitPositionConstraint(const RigidBodyTree<double>& tree,
+                              string body_name,
+                              Vector3d translation,
+                              Vector3d desired_pos) :
+    DirconAbstractConstraint<double>(3,
+                                     tree.get_num_positions(),
+                                     VectorXd::Zero(3),
+                                     VectorXd::Zero(3),
+                                     body_name + "_position_constraint"),
+    tree_(tree),
+    body_idx_(multibody::GetBodyIndexFromName(tree, body_name)),
+    translation_(translation),
+    desired_pos_(desired_pos) {
+  }
+  ~BodyPonitPositionConstraint() override = default;
+
+  void EvaluateConstraint(const Eigen::Ref<const drake::VectorX<double>>& x,
+                          drake::VectorX<double>* y) const override {
+    VectorXd q = x;
+    KinematicsCache<double> cache = tree_.doKinematics(q);
+    VectorXd pt = tree_.transformPoints(cache, translation_, body_idx_, 0);
+
+    *y = pt - desired_pos_;
+  };
+ private:
+  const RigidBodyTree<double>& tree_;
+  const int body_idx_;
+  const Vector3d translation_;
+  const Vector3d desired_pos_;
+};
+
+// Use fixed-point solver to get configuration guess (for standing in place)
+void GetInitFixedPointGuess(const Vector3d& pelvis_position,
+                            const RigidBodyTree<double>& tree,
+                            VectorXd* q_init,
+                            VectorXd* u_init,
+                            VectorXd* lambda_init) {
+  int n_q = tree.get_num_positions();
+  int n_v = tree.get_num_velocities();
+  int n_u = tree.get_num_actuators();
+
+  int toe_left_idx = GetBodyIndexFromName(tree, "toe_left");
+  int toe_right_idx = GetBodyIndexFromName(tree, "toe_right");
+  Vector3d pt_front_contact(-0.0457, 0.112, 0);
+  Vector3d pt_rear_contact(0.088, 0, 0);
+  MatrixXd xa(3, 4);
+  xa.col(0) = pt_front_contact;
+  xa.col(1) = pt_rear_contact;
+  xa.col(2) = pt_front_contact;
+  xa.col(3) = pt_rear_contact;
+  std::vector<int> idxa;
+  idxa.push_back(toe_left_idx);
+  idxa.push_back(toe_left_idx);
+  idxa.push_back(toe_right_idx);
+  idxa.push_back(toe_right_idx);
+  ContactInfo contact_info(xa, idxa);
+
+  VectorXd q_desired = VectorXd::Zero(n_q);
+  q_desired << -0.0872062,
```

Comment:
Where did these magic numbers come from (comment would be useful). Can these be rounded?

Reply:
It came from inverse kinematics where cassie was standing on the ground. 
I also added documentation and rounded some of the numbers.